nodejs/server.js

```diff
@@ -167,4 +169,116 @@ app.all('/', function (req, res) {
 
 });
 
-app.listen(argv.port);
+let server = require('http').createServer({
+    perMessageDeflate: {
+        zlibDeflateOptions: {
+            // See zlib defaults.
+            chunkSize: 1024,
+            memLevel: 7,
+            level: 3
+        },
+        zlibInflateOptions: {
+            chunkSize: 10 * 1024
+        },
+        // Other options settable:
+        clientNoContextTakeover: true, // Defaults to negotiated value.
+        serverNoContextTakeover: true, // Defaults to negotiated value.
+        serverMaxWindowBits: 10, // Defaults to negotiated value.
+        // Below options specified as default values.
+        concurrencyLimit: 10, // Limits zlib concurrency for perf.
+        threshold: 1024 // Size (in bytes) below which messages
+        // should not be compressed.
+    }
+});
+
+// Also mount the app here
+server.on('request', app);
+
+const websocketEvent = {
+    url: 'http://localhost:8000/websocket'
```

Comment:
What kind of throughput can the events system underlying this support? I think new connection events might be just as frequent as requests for some applications -- so is k8s fine with many many of these events?

Reply:
There are two aspects to it:
* Kubernetes itself holds events upto 1hr and etcd needs to be scaled when and if required. Fission could provide metrics about number of events to help take such a decision. 
* Fission's component processing those events. We can plan on making the particular component scalable if required . 

Also, we could probably limit the generation of event to only the first websocket connection. WDYT?

Reply:
Yeah, I think just sending the first event makes a lot of sense! Did you folks use events instead of RPC to the executor or something like that because you get a buffer and queuing?

Reply:
It was mostly to decouple the components.